File: Java_Basic_Jan_2023_SolutionsOfProblems/P03AdditionalProblems_ConditionalStatementsAdvanced/P02_BikeRace.java
package P03AdditionalProblems_ConditionalStatementsAdvanced;

import java.util.Scanner;

public class P02_BikeRace {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numberJuniorBikers = Integer.parseInt(scanner.nextLine());
        int numberSeniorBikers = Integer.parseInt(scanner.nextLine());
        String routeType = scanner.nextLine();
        double donationSum = 0;

        switch (routeType) {
            case "trail":
                donationSum = numberJuniorBikers * 5.5 + numberSeniorBikers * 7.0;
                break;
            case "cross-country":
                donationSum = numberJuniorBikers * 8.0 + numberSeniorBikers * 9.5;
                if ((numberJuniorBikers + numberSeniorBikers) >= 50) {
                    donationSum = donationSum * 0.75;
                }
                break;
            case "downhill":
                donationSum = numberJuniorBikers * 12.25 + numberSeniorBikers * 13.75;
                break;
            case "road":
                donationSum = numberJuniorBikers * 20.0 + numberSeniorBikers * 21.5;
                break;
        }
        donationSum = donationSum * 0.95;
        System.out.printf("%.2f", donationSum);
    }
}

File: JavaScript_Basic_04_2023_SolutionsOfProblems/03.MoreEx_PB-JS-Nested-Conditional-Statements/P02_BikeRace.js
function bikeRace(input) {
    let juniorsNumber = Number(input[0]);
    let seniorsNumber = Number(input[1]);
    let routeType = input[2];

    let donatedSum = 0.0;

    switch (routeType) {
        case "trail":
            donatedSum = juniorsNumber * 5.5 + seniorsNumber * 7;
            break;
        case "cross-country":
            donatedSum = juniorsNumber * 8 + seniorsNumber * 9.5;
            if ((juniorsNumber + seniorsNumber) >= 50) {
                donatedSum = donatedSum * 0.75;
            }
            break;
        case "downhill":
            donatedSum = juniorsNumber * 12.25 + seniorsNumber * 13.75;
            break;
        case "road":
            donatedSum = juniorsNumber * 20 + seniorsNumber * 21.5;
            break;
    }

    donatedSum = donatedSum * 0.95;
    console.log(donatedSum.toFixed(2));
}

bikeRace([30,
    25,
    "cross-country" 
    
    
    ])